packages/eslint-plugin/src/utils/configHelpers.js

```diff
@@ -131,39 +131,50 @@ module.exports = {
       return [];
     }
 
-    /**
-       * Note that this way of reading tsconfig does not account for extends, but that's okay since
-       * typically include will not be specified that way
-       *
-       * @type {{
-          extends: string[];
-          include: string[];
-          excludes: string[];
-          compilerOptions: Record<string,unknown>;
-          references?: Array<{path:string}>
-          }}
-       */
-    const tsconfig = jju.parse(fs.readFileSync(tsconfigPath).toString());
+    const ts = require('typescript');
```

Comment:
Can you add a comment that this is delay loaded so that typescript is only required/parsed when needed? (without a comment it's likely to get moved up to the top of the file at some point)